internal/pkg/application/conversion/converters.go

```diff
@@ -4,88 +4,169 @@ import (
 	"context"
 	"fmt"
 	"math"
+	"strings"
 	"time"
 
 	"github.com/diwise/iot-agent/internal/pkg/application/decoder/payload"
-	"github.com/diwise/iot-core/pkg/lwm2m"
-	"github.com/diwise/iot-core/pkg/measurements"
 	"github.com/farshidtz/senml/v2"
 )
 
-type MessageConverterFunc func(ctx context.Context, internalID string, p payload.Payload) (senml.Pack, error)
+type MessageConverterFunc func(ctx context.Context, internalID string, p payload.Payload, fn func(p senml.Pack) error) error
 
-func Temperature(ctx context.Context, deviceID string, p payload.Payload) (senml.Pack, error) {
-	if temp, ok := payload.Get[float64](p, measurements.Temperature); ok {
-		return NewSenMLPack(deviceID, lwm2m.Temperature, p.Timestamp(), Value(measurements.Temperature, temp)), nil
+func Temperature(ctx context.Context, deviceID string, p payload.Payload, fn func(p senml.Pack) error) error {
+	/*
+		ObjectURN: urn:oma:lwm2m:ext:3303
+		ID      Name            Type     Unit
+		5700    Sensor Value    Float
+	*/
+
+	if temp, ok := payload.Get[float64](p, "temperature"); ok {
+		pack := NewSenMLPack(deviceID, "urn:oma:lwm2m:ext:3303", p.Timestamp(), Value("5700", temp))
```

Comment:
I would prefer local constants instead of this "comment plus repetition of value" (that I might need to go back and look up in the comment to understand). IMHO readability would be improved with const Temperature string = "urn:oma:lwmwm:ext:3303" and const SensorValue string = "5700" instead of the comment.

Reply:
and then something like NewSenMLPack(deviceID, Temperature, p.Timestamp(), Value(SensorValue, temp))